eclipse-collections/src/main/java/org/eclipse/collections/impl/list/primitive/IntInterval.java

```diff
@@ -251,6 +251,10 @@ public static IntInterval fromToBy(int from, int to, int stepBy)
         {
             throw new IllegalArgumentException("Step by is incorrect for the range");
         }
+        if ((from > to && (from + stepBy < to)) || (from < to  && (from + stepBy > to)))
```

Comment:
this is still wrong, because you can't use addition without overflow. The correct logic for the forward case is:
```(from < to) && (to - from < stepBy)```
Test case: see below

Reply:
@donraab this still needs to be fixed.

Reply:
also the reverse case needs to be fixed.

Reply:
Working on this now.